src/lib/blueprint.ts

```diff
@@ -117,14 +121,327 @@ interface ObjectProperty extends BaseProperty {
+
+  return {
+    name:
+      'operationId' in operation && typeof operation.operationId === 'string'
+        ? operation.operationId
+        : `${path.replace(/\//g, '')}${method.charAt(0).toUpperCase() + method.slice(1).toLowerCase()}`,
+    path: endpointPath,
+    description:
+      'description' in operation && typeof operation.description === 'string'
+        ? operation.description
+        : '',
+    isUndocumented: false,
+    isDeprecated: false,
+    deprecationMessage: '',
+    parameters: createParameters(operation),
+    request: createRequest(method, operation),
+    response: createResponse(
+      'responses' in operation ? operation.responses : {},
+    ),
+  }
+}
+
+const createParameters = (operation: OpenAPIOperation): Parameter[] => {
+  return 'parameters' in operation && Array.isArray(operation.parameters)
+    ? operation.parameters
+        .filter((param) => typeof param === 'object' && param !== null)
+        .map(createParameter)
+    : []
+}
+
+const createParameter = (param: OpenAPIParameter): Parameter => {
+  return {
+    name: 'name' in param && typeof param.name === 'string' ? param.name : '',
+    isRequired:
+      'required' in param && typeof param.required === 'boolean'
+        ? param.required
+        : false,
+    isUndocumented: false,
+    isDeprecated: false,
+    deprecationMessage: '',
+    description:
+      'description' in param && typeof param.description === 'string'
+        ? param.description
+        : '',
+  }
+}
+
+const createRequest = (
+  method: string,
+  operation: OpenAPIOperation,
+): Request => {
+  const uppercaseMethod = openapiMethodToMethod(
+    method.toLowerCase() as LowercaseMethod,
+  )
+
+  return {
+    methods: [uppercaseMethod],
+    semanticMethod: uppercaseMethod,
+    preferredMethod: uppercaseMethod,
+    parameters: createParameters(operation),
+  }
+}
+
+const createResources = (
+  schemas: OpenAPI['components']['schemas'],
+  isFakeData: boolean,
+  targetSchema: string,
+): Record<string, Resource> => {
+  return Object.entries(schemas)
+    .filter(([schemaName]) => isFakeData || schemaName === targetSchema)
+    .reduce<Record<string, Resource>>((acc, [schemaName, schema]) => {
+      if (
+        typeof schema === 'object' &&
+        schema !== null &&
+        'properties' in schema &&
+        typeof schema.properties === 'object' &&
+        schema.properties !== null
+      ) {
+        acc[schemaName] = {
+          resourceType: schemaName,
+          properties: createProperties(schema.properties),
+        }
+      }
+      return acc
+    }, {})
+}
+
+const createResponse = (responses: OpenAPIOperation['responses']): Response => {
+  if (typeof responses !== 'object' || responses === null) {
+    return { responseType: 'void', description: 'No description available' }
+  }
+
+  const okResponse = responses['200']
+  if (typeof okResponse !== 'object' || okResponse === null) {
```

Comment:
I noticed a lot of checks for `(typeof foo !== 'object' || foo === null)`. Are there cases in the openapi schema where some property can be null, object, or something else, like string or bool? 

Otherwise, can we simplify all of these checks to eiter `foo === null` of `foo == null`?

Reply:
I've tried that but in most cases it will throw out: `foo can be possibly undefined`

Reply:
Hmm, if you do `foo != null` or `foo == null` as the two kinds of checks (without the `typeof`) it should work, e.g., just `if (okResponse == null) {`